### ISSUE TITLE
Invalid format when saving PSBT files

### ISSUE THREAD

**Issue by Quantris** [bitcoin/bitcoin#20959]
As referenced @ https://bitcoin.stackexchange.com/questions/101611/bitcoin-core-0-21-0-loading-psbt-from-file-gets-error-unable-to-decode-psbt

I was able to reproduce the issue described, by saving & then immediately loading a PSBT file from my Bitcoin Core instance which has a watch-only wallet loaded.

I get the "global unsigned tx key is more than one byte" error using a file saved from the dialog that shows up after Send -> Create Unsigned

The PSBT on clipboard is OK and I can load it successfully via File -> Load PSBT from clipboard
The dialog that opens there offers the option to save to file. The file saved via that approach gives a different error when I try to load it: "extra data after PSBT"

This is on Windows so I suspect this is related to text vs. binary encoding messing with CRLF
In particular these ofstreams:
https://github.com/bitcoin/bitcoin/blob/cb2c578451770a29395f79f7077c362e1f54dfdd/src/qt/sendcoinsdialog.cpp#L437
https://github.com/bitcoin/bitcoin/blob/c7ad94428ab6f54661d7a5441e1fdd0ebf034903/src/qt/psbtoperationsdialog.cpp#L148

should be created in binary mode (the default is text mode)

**Comment by fanquake** [comment]
cc @achow101 @instagibbs

**Comment by achow101** [comment]
I have seen this compliant from another person as well, and from the sample PSBTs he has given me, it does appear to be the case. It looks like every `0x0a` (newline character) is accompanied by a `0x0d` (carriage return character). So the problem most likely is due to the text mode and Windows using CRLF instead of just LF.

**Comment by practicalswift** [comment]
Excellent bug report @Quantris! Thanks for taking the time to report this.

Welcome as a contributor to the project!

**Comment by perkio** [comment]
I encountered the same issue using Bitcoin Core v0.21.0 on Windows. 

Whenever I tried to create an unsigned PSBT file for signing on a hardware wallet, the wallet would reject the file. I found out that when saving a binary PSBT to disk the file contains CRLF (`0x0a`+`0x0d`) for new line separation *but* the clipboard PSBT Base64 version which is generated prior to saving the PSBT file is valid and has LF (`0x0a`) as expected. (This correlates with text/binary mode finding stated above.)

So, If anyone else is running into this, a quick workaround until a patch is available would be to save the binary by decoding the Base64 PSBT.
